### PR TITLE
adding ability to pull from different directories for test data & answer files

### PR DESCRIPTION
**Pull request by sambish5**
creates getMyTestAnswerFiles, which allows for test answer files from one directory to work with data files from another directory.

### CODE REVIEW

File: src/test/java/emissary/test/core/junit5/UnitTest.java [hunk 1]
```diff
@@ -199,7 +222,19 @@ protected Document getAnswerDocumentFor(String resource) {
             return null;
         }
 
-        String aname = resource.substring(0, datPos) + ResourceReader.XML_SUFFIX;
+        String aname = "";
+        if (answerFiles.isEmpty()) {
+            aname = resource.substring(0, datPos) + ResourceReader.XML_SUFFIX;
+        } else {
+            // if answer files are in different directory than data files, this will be used to find matching answer to data pair
+            int testFileName = resource.lastIndexOf("/");
```
Comment (dev-mlb) [inline src/test/java/emissary/test/core/junit5/UnitTest.java:230]:
```suggestion
           String testFileName = FilenameUtils.getBaseName(resource);
```

File: src/test/java/emissary/test/core/junit5/UnitTest.java [hunk 2]
```diff
@@ -199,7 +222,19 @@ protected Document getAnswerDocumentFor(String resource) {
...
+            for (String answer : answerFiles) {
+                if (answer.contains(resource.substring(testFileName + 1, datPos))) {
```
Comment (dev-mlb) [inline src/test/java/emissary/test/core/junit5/UnitTest.java:232]:
```suggestion
                if (FilenameUtils.getBaseName(answer).equals(testFileName)) {
```

File: src/test/java/emissary/test/core/junit5/UnitTest.java (hunk 1 above)
Comment (dev-mlb) [inline src/test/java/emissary/test/core/junit5/UnitTest.java:230]:
```suggestion
            String testFileName = FilenameUtils.getBaseName(resource);
```

File: src/test/java/emissary/test/core/junit5/UnitTest.java [hunk 3]
```diff
@@ -132,6 +141,20 @@ public static Stream<? extends Arguments> getMyTestParameterFiles(Class<?> clz)
...
     }
 
+    /**
+     * Get test resources (*.dat) and test answers when they are in two different directories.
+     * 
+     * @param answerClass class that provides the test answer files
+     * @param dataClass class that provides the test resource (*.dat) files
+     * @return the stream of test resource files to be used for JUnit Parameterized Tests
+     */
+    public static Stream<? extends Arguments> getMyTestAnswerFiles(Class<?> answerClass, Class<?> dataClass) {
```
Comment (dev-mlb) [inline src/test/java/emissary/test/core/junit5/UnitTest.java:151]: Maybe something more like this?? --
```
    public static Stream<? extends Arguments> getMyTestParameterFiles(Class<?> clz) {
        return getMyTestParameterFiles(clz, clz);
    }

    public static Stream<? extends Arguments> getMyTestParameterFiles(Class<?> clz, Class<?> ansClz) {
        ResourceReader rr = new ResourceReader();
        List<String> rs = rr.findDataResourcesFor(clz);
        answerFiles = getMyTestAnswerFiles(rr, ansClz);
        return rs.stream().map(Arguments::of);
    }

    private static List<String> getMyTestAnswerFiles(ResourceReader rr, Class<?> answerClass) {
        return rr.findXmlResourcesFor(answerClass);
    }
```

**Review by dev-mlb — CHANGES_REQUESTED** [review]

**Review by dev-mlb — APPROVED** [review]

File: src/test/java/emissary/test/core/junit5/UnitTest.java [hunk 4]
```diff
@@ -199,7 +225,19 @@ protected Document getAnswerDocumentFor(String resource) {
...
-        String aname = resource.substring(0, datPos) + ResourceReader.XML_SUFFIX;
+        String aname = "";
+        if (answerFiles.isEmpty()) {
+            aname = resource.substring(0, datPos) + ResourceReader.XML_SUFFIX;
+        } else {
+            // if answer files are in different directory than data files, this will be used to find matching answer to data pair
+            String testFileName = FilenameUtils.getBaseName(resource);
+            for (String answer : answerFiles) {
+                if (FilenameUtils.getBaseName(answer).equals(testFileName)) {
+                    aname = answer;
```
Comment (fbruton) [inline src/test/java/emissary/test/core/junit5/UnitTest.java:236]: Suggest we break out after finding a match.
```suggestion
                    aname = answer;
                    break;
```

**Review by fbruton — CHANGES_REQUESTED** [review]

**Review by jpdahlke — APPROVED** [review]

**Review by fbruton — APPROVED** [review]

**Pull request merged**